sql/plan/bindvar.go

```diff
@@ -92,6 +92,31 @@ func applyBindingsHelper(n sql.Node, bindings map[string]sql.Expression) (sql.No
 			}
 			ne, _, err := transform.NodeExprs(n.WithSource(newSource), fixBindingsTransform)
 			return ne, transform.NewTree, err
+		case *ResolvedTable:
```

Comment:
Seems like calling these ResolvedTable is misleading, like they will panic if you try to read them. I know we talked about something like `DeferredFilteredTable`, is that not an easy add to avoid the confusion?

Reply:
It's definitely possible, but I think it is definitely a bit harder.

DeferredTable is created a in `resolveTable` (in BeforeDefault), but during that time filters haven't yet been evaluated, so I can't save them there.
I could add a new rule right before pushdownFilters? and then `applyBindings` might just work?
or I could add a new rule/modify pushdownFilters (in AfterDefault), to create a DeferredTable there?

Reply:
`WithFilters` has one use in pushdown. I think a helper function should look for a `BindVar` in the `WithFilters` argument, and we switch on the result to make either a `ResolvedTable` or `DeferredFiltersTable`. Deferred table will implemented `UnresolvedTable` and look a lot like `DeferredAsOf`, and will be a target for bindvar replacement.